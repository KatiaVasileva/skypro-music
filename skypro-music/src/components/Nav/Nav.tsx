"use client";

import Image from "next/image";
import styles from "./Nav.module.css";
import React, { useState } from "react";
import { useAppDispatch, useAppSelector } from "@/store/store";
import { setIsMyPlaylistClicked } from "@/store/features/trackSlice";
import { getAccessTokenFromLocalStorage } from "@/utils/helpers";
import { useRouter } from "next/navigation";

function Nav() {
  const dispatch = useAppDispatch();
  const refreshToken = useAppSelector((state) => state.user.tokens?.refresh);
  const [isBurgerClicked, setIsBurgerClicked] = useState(false);
  const router = useRouter();

  const handleBurgerClick: React.MouseEventHandler<HTMLDivElement> = () => {
    setIsBurgerClicked((prevState) => !prevState);
  };

  const handleMainClick: React.MouseEventHandler<HTMLAnchorElement> = (
    event
  ) => {
    event.preventDefault();
    dispatch(setIsMyPlaylistClicked(false));
    router.push("/");
  };

  const handleMyPlaylistClick: React.MouseEventHandler<
    HTMLAnchorElement
  > = async (event) => {
    event.preventDefault();
    const access = getAccessTokenFromLocalStorage();
    if (!access) {
      alert("Необходимо зарегистрироваться");
      return;
    }
    dispatch(setIsMyPlaylistClicked(true));
    router.push("/favorite");
  };

  return (
    <nav className={styles.main}>
      <div className={styles.logo}>
        <Image
          className={styles.image}
          src="/img/logo.png"
          alt="logo"
          width={113.33}
          height={43}
        />
      </div>
      <div className={styles.burger} onClick={handleBurgerClick}>
        <span className={styles.burgerLine}></span>
        <span className={styles.burgerLine}></span>
        <span className={styles.burgerLine}></span>
      </div>
      {isBurgerClicked && (
        <div className={styles.menu}>
          <ul className={styles.menuList}>
            <li className={styles.menuItem}>
              <a href="#" className={styles.menuLink} onClick={handleMainClick}>
                Главное
              </a>
            </li>
            <li className={styles.menuItem}>
              <a
                href=""
                className={styles.menuLink}
                onClick={handleMyPlaylistClick}
              >
                Мой плейлист
              </a>
            </li>
            <li className={styles.menuItem}>
              <a href="/signin" className={styles.menuLink}>
                {/* {user ? "Выйти" : "Войти"} */}
                Войти
              </a>
            </li>
          </ul>
        </div>
      )}
    </nav>
  );
}

export default Nav;
