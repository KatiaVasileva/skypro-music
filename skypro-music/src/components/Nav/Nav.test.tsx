import ReduxProvider from "@/store/ReduxProvider";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Nav from "./Nav";

describe("Nav component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ReduxProvider>
        <Nav />
      </ReduxProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
