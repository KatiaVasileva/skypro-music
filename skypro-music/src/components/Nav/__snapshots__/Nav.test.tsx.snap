// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders correctly 1`] = `
<div>
  <nav
    class="main"
  >
    <div
      class="logo"
    >
      <a
        href="/playlist"
      >
        <img
          alt="logo"
          class="image"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="43"
          src="/_next/image?url=%2Fimg%2Flogo.png&w=256&q=75"
          srcset="/_next/image?url=%2Fimg%2Flogo.png&w=128&q=75 1x, /_next/image?url=%2Fimg%2Flogo.png&w=256&q=75 2x"
          style="color: transparent;"
          width="113.33"
        />
      </a>
    </div>
    <div
      class="burger"
    >
      <span
        class="burgerLine"
      />
      <span
        class="burgerLine"
      />
      <span
        class="burgerLine"
      />
    </div>
  </nav>
</div>
`;
